### ISSUE TITLE
Gvalue and Get/Set property

### ISSUE THREAD

**Issue by hythm7** [Xliff/p6-GtkPlus#24]
How to get or set properties of widgets? I have the below code but I can't figure out how to use GValue:

```
use GTK::Application;
use GTK::Compat::Types;
use GTK::Box;
use GTK::Button;
use GTK::Grid;
#use NativeCall;

my $app = GTK::Application.new(
  title  => 'org.genex.test.widget',
  width  => 400,
  height => 400
);

$app.activate.tap({
  
my GTK::Button $exit .= new_with_label: <exit>;
my GTK::Button $swap .= new_with_label: <swap>;
my GTK::Button $hello .= new_with_label: <hello>;
my GTK::Button $world .= new_with_label: <world>;

$exit.clicked.tap: { $app.exit  };

my GTK::Grid $grid .= new;

$grid.attach: $hello, 0, 0, 1, 1;
$grid.attach: $world, 1, 0, 1, 1;

#my $v = GValue.new;

$swap.clicked.tap: {
  $grid.child-set-property($hello, 'left_attach', 1);
  $grid.child-set-property($world, 'left_attach', 0);
};

my $box = GTK::Box.new-vbox(4);

$box.pack_start($grid, False, True, 0);
$box.pack_start($swap, False, True, 0);
$box.pack_start($exit, False, True, 0);

$app.window.add: $box;
$app.show_all;
});

$app.run;

```

**Comment by Xliff** [comment]
That's a tricky problem. For this, you don't need GValues, although if you want a tutorials on  those, I'd be happy to give it.

No, the problem here is that there are several types of child properties and there are get-set methods for each type. Currently the following GTK::Container methods exist:

```
child-get-bool
child-set-bool
child-get-int
child-set-int
child-get-uint
child-set-uint
```

For your code, you'd want to use something like the following:

```perl6
$swap.clicked.tap: {
  $grid.child-set-int($hello, 'left-attach', 1);
  $grid.child-set-int($world, 'left-attach', 0);
};
```

Please note that GTK properties use dashes and not underscores. Most functions have proper aliases, but in this case we need strings, so they must match exactly.

**Comment by hythm7** [comment]
My bad, I should have used `child-set-int` instead of `child-set-property`. Closing

**Comment by Xliff** [comment]
LOL! No problem. I just finished writing a comment regarding that. 

Do you have a repository that includes the examples you are writing? I'd love to take a look.

**Comment by hythm7** [comment]
Just saw the comment, Thank you for the explanation. 

I don't have a repository for the examples.  I'm learning GTK by working on hobby project https://github.com/hythm7/Z , and when I have an issue I simplify the problem to fit as an example. 

Creating a separate repository for examples is a good idea, I can convert the things I learned and implemented in the above project "Z" to examples, and add new ones along this way.

Unfortunately I'm still at the  beginning in learning GTK (Obvious from the issues I submitted), so I expect adding the examples will be slow process. 

Once I create and have something in the examples repository, I will send you the link.
